cloud/chatevent.js

```diff
@@ -59,13 +59,24 @@ function createPresence(alias, subtype) {
     return presence.save();
 }
 
-function getMostRecentForChatRoom(chatRoomId) {
+function getMostRecentForChatRoom(chatRoomId, alias) {
     var aliasQuery = new Parse.Query("Alias");
     aliasQuery.equalTo("chatRoomId", chatRoomId);
 
     var query = new Parse.Query("ChatEvent");
     query.matchesQuery("alias", aliasQuery);
     query.descending("createdAt");
     query.include("alias");
-    return query.first();
+    
+    var deletedChatEventIds = alias.get("deletedChatEventIds");
```

Comment:
I think there's a firstclass Parse query constraint equivalent of this. This is fine for now but can you add a comment that marks this for refactoring